packages/grid/x-data-grid-pro/src/components/GridColumnHeaders.tsx

```diff
@@ -267,6 +274,11 @@ const GridColumnHeaders = React.forwardRef<HTMLDivElement, DataGridProColumnHead
             minFirstColumn: leftPinnedColumns.length,
             maxLastColumn: visibleColumnFields.length - rightPinnedColumns.length,
           })}
+          {getColumnFilters({
```

Comment:
There's a slight layout jump when navigating between filter row and column headers:

https://github.com/mui/mui-x/assets/13808724/d734d705-8b19-4968-b465-b2648e69efda

Reply:
Interesting. It's not reproducible on Firefox but on Chrome, and the same is also observed with [column grouping](https://mui.com/x/react-data-grid/column-groups/#column-reordering).

![groupedColumns](https://github.com/mui/mui-x/assets/12609561/cfde6cab-7a43-4247-af0a-50bcc412c9c9)

Seems like a CSS thing, will investigate more. Thanks for reporting.